File: 11-ABCheck/11.hs
--Is any part of a string in the form of a???b or b???a ?
abCheck :: String -> Bool

abCheck "" = False
abCheck ('a':x:y:z:'b':rest) = True
abCheck ('b':x:y:z:'a':rest) = True
abCheck (x:xs) = abCheck xs

main = do
  print $ abCheck "adkeb"   == True
  print $ abCheck "sadkaib" == False

File: 11-ABCheck/11.js
// Is any part of a string in the form of a???b or b???a ?
function ABCheck(str) { 
  return /(a...b)|(b...a)/gi.test(str)    
}

function ABCheckInvolved(str){
  for(var i=0; i<str.length; i++){
    if(str[i] === "a" && str[i+4] === "b" || str[i] === "b" && str[i+4] === "a"){
      return true;
    }
  }

  return false;
}

console.log(ABCheckInvolved("adkeb")  === true)
console.log(ABCheckInvolved("sadkaib") === false)   
